### PR TITLE
Invalidate ActualFiniteStates while editing an ActualFiniteStateList; fixes #335

### LINKED ISSUE #335: Invalidate ActualState list in ActualStateList dialog when updating linked PossibleStateLists
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the CDP4-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
In two following scenarios the list of ActualStates located in the ActualStateListDialog should have updated behavior:

- Creating the ActualStateList: The list should either not be visible at all as it will always be empty, or provide a preview (clearly stating it is a preview) of the actual finite states that will be created. In the second case the StateKind selection should be disabled.

- Updating the ActualSateList: The list should become visibly disabled to show that is no longer valid is the list of PossibleFiniteStates is changed in order or updated in any way. If Preview is made possible for the create case, this can be also done for the update and visibly indicated as so. StateKind change should be disabled.

### Logs

  - [ ] There are no logs regarding the issue
  - { ] I have attached a log file or included the relevant section in this issue  
  
### System Configuration

- CDP4 version:
  - [ ] CDP4-IME: 
  - [ ] Other: 
  -
- Web Services:
  - [ ] CDP4 Web Services version: 
  - [ ] OCDT WSP version: 
- Environment (Operating system, version and so on):
  - [ ] Microsoft Excel (32/64 bit): 
  - [ ] Windows OS (32/64 bit): 
  - [ ] Additional information:

### PR DESCRIPTION
**Pull request by nathanatrhea**
When Creating/Editing ActualFiniteStateLists the ActualFiniteStates can be invalidated; fixes #335

# Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
When Creating/Editing ActualFiniteStateLists the ActualFiniteStates can be invalidated; fixes #335

### CODE REVIEW

File: EngineeringModel/Views/Dialogs/ActualFiniteStateListDialog.xaml [hunk 1]
```diff
@@ -13,6 +13,7 @@
              xmlns:cdp4Composition="clr-namespace:CDP4Composition;assembly=CDP4Composition"
              xmlns:dxe="http://schemas.devexpress.com/winfx/2008/xaml/editors"
              xmlns:engineeringModelData="clr-namespace:CDP4Common.EngineeringModelData;assembly=CDP4Common"
+             xmlns:xtraEditors="clr-namespace:DevExpress.XtraEditors;assembly=DevExpress.XtraEditors.v19.2"
```
Comment (lxatrhea) [inline EngineeringModel/Views/Dialogs/ActualFiniteStateListDialog.xaml]: Is this extra reference necessary?

Comment (nathanatrhea) [reply EngineeringModel/Views/Dialogs/ActualFiniteStateListDialog.xaml]: Good point it has been added while I was trying to achieve something which is not required at last

File: EngineeringModel/Views/Dialogs/ActualFiniteStateListDialog.xaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿<dx:DXWindow x:Class="CDP4EngineeringModel.Views.ActualFiniteStateListDialog"
+<dx:DXWindow x:Class="CDP4EngineeringModel.Views.ActualFiniteStateListDialog"
```
Comment (lxatrhea) [inline EngineeringModel/Views/Dialogs/ActualFiniteStateListDialog.xaml]: I think there is still something wrong with encoding in your git settings. I see a lot of first rows of code files that have a indication of being changed, but the string/rows seem exactly the same.

Comment (nathanatrhea) [reply EngineeringModel/Views/Dialogs/ActualFiniteStateListDialog.xaml]: this bothers me. it makes dirty commits

File: EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs [hunk 3]
```diff
@@ -48,6 +48,25 @@ public class ActualFiniteStateListDialogViewModel : CDP4CommonView.ActualFiniteS
...
+        /// </summary>
+        /// <remarks>
+        /// The value shall be set to false when any change is made on the possible finite state list
+        /// </remarks>
+        public bool IsValueSetEditable
+        {
+            get => this.isValueSetEditable && base.IsNonEditableFieldReadOnly;
+            private set => this.RaiseAndSetIfChanged(ref this.isValueSetEditable, value);
+        }
+
```
Comment (alexatrhea) [inline EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs]: There's 3 empty lines in a row here.

**Comment by alexatrhea** [comment]
One big thing is that the docs and the variable names keep refering to Value Sets. This is incorrect. A `ValueSet` in 10-25 means something very specific. Here you are invalidating `ActualFiniteStates` that are in an `ActualFiniteStateList` when any changes are made to a `PossibleFiniteStateList` or any of the contained `PossibleFiniteStates`. Please refactor the variable naming and the docs to reflect this.

File: EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs [hunk 4]
```diff
@@ -48,6 +48,23 @@ public class ActualFiniteStateListDialogViewModel : CDP4CommonView.ActualFiniteS
...
         /// </summary>
         private ReactiveList<Option> includeOption;
+        
+        /// <summary>
+        /// Backing field for <see cref="IsActualFiniteStatesListValid"/>
+        /// </summary>
+        private bool isActualFiniteStatesListValid;
+
+        /// <summary>
+        /// Gets a value indicating whether the ActualFiniteStateList needs to be re-calculated or not
```
Comment (alexatrhea) [inline EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs]: `Gets a value indicating whether the ActualFiniteStateList needs to be re-calculated or not`

becomes

`Gets a value indicating whether the ActualFiniteStates of this ActualFiniteStateList needs to be re-calculated or not`

File: EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs [hunk 5]
```diff
@@ -48,6 +48,23 @@ public class ActualFiniteStateListDialogViewModel : CDP4CommonView.ActualFiniteS
...
+        /// </summary>
+        /// <remarks>
+        /// The value shall be set to false when any change is made on the possible finite state list
+        /// </remarks>
+        public bool IsActualFiniteStatesListValid
```
Comment (alexatrhea) [inline EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs]: `IsActualFiniteStatesListValid` -> `AreActualFiniteStatesValid` please

File: EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs [hunk 6]
```diff
@@ -48,6 +48,23 @@ public class ActualFiniteStateListDialogViewModel : CDP4CommonView.ActualFiniteS
         /// Backing field for <see cref="ExcludeOption"/>s
         /// </summary>
         private ReactiveList<Option> includeOption;
+        
+        /// <summary>
+        /// Backing field for <see cref="IsActualFiniteStatesListValid"/>
+        /// </summary>
+        private bool isActualFiniteStatesListValid;
```
Comment (alexatrhea) [inline EngineeringModel/ViewModels/Dialogs/ActualFiniteStateListDialogViewModel.cs]: `isActualFiniteStatesListValid` -> `areActualFiniteStatesValid` please

**Pull request merged**